src/server.c:14:10: fatal error: 'wlr/types/wlr_gtk_primary_selection.h' file not found
#include <wlr/types/wlr_gtk_primary_selection.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/view.c:812:47: error: no member named 'subsurfaces' in 'struct wlr_surface'
  wl_list_for_each (wlr_subsurface, &surface->subsurfaces, parent_link) {
                                     ~~~~~~~  ^

--- src/server.c.orig	2021-04-09 17:37:23 UTC
+++ src/server.c
@@ -11,7 +11,6 @@
 #include <wlr/types/wlr_compositor.h>
 #include <wlr/types/wlr_data_control_v1.h>
 #include <wlr/types/wlr_data_device.h>
-#include <wlr/types/wlr_gtk_primary_selection.h>
 #include <wlr/types/wlr_input_device.h>
 #include <wlr/types/wlr_keyboard.h>
 #include <wlr/types/wlr_output_layout.h>
@@ -610,7 +609,6 @@ setup_selection(struct hikari_server *server)
 {
   wlr_data_control_manager_v1_create(server->display);
 
-  wlr_gtk_primary_selection_device_manager_create(server->display);
   wlr_primary_selection_v1_device_manager_create(server->display);
 
   server->seat = wlr_seat_create(server->display, "seat0");
--- src/view.c.orig	2021-04-09 17:37:23 UTC
+++ src/view.c
@@ -809,7 +809,13 @@ hikari_view_map(struct hikari_view *view, struct wlr_s
   wl_signal_add(&surface->events.new_subsurface, &view->new_subsurface);
 
   struct wlr_subsurface *wlr_subsurface;
-  wl_list_for_each (wlr_subsurface, &surface->subsurfaces, parent_link) {
+  wl_list_for_each (wlr_subsurface, &surface->subsurfaces_below, parent_link) {
+    struct hikari_view_subsurface *subsurface =
+        (struct hikari_view_subsurface *)malloc(
+            sizeof(struct hikari_view_subsurface));
+    hikari_view_subsurface_init(subsurface, view, wlr_subsurface);
+  }
+  wl_list_for_each (wlr_subsurface, &surface->subsurfaces_above, parent_link) {
     struct hikari_view_subsurface *subsurface =
         (struct hikari_view_subsurface *)malloc(
             sizeof(struct hikari_view_subsurface));
